src/data_type/mod.rs

```diff
@@ -2285,6 +2285,47 @@ impl DataType {
             (Err(_), Err(_)) => Err(Error::other("No common variant")),
         }
     }
+
+    pub fn all_values(&self) -> bool {
+        match &self {
+            DataType::Integer(i) => i.all_values(),
+            DataType::Float(f) => f.all_values(),
+            DataType::Date(d) => d.all_values(),
+            DataType::Time(t) => t.all_values(),
+            DataType::DateTime(d) => d.all_values(),
+            DataType::Duration(d) => d.all_values(),
+            DataType::Boolean(b) => b.all_values(),
+            DataType::Text(t) => t.all_values(),
+            DataType::Optional(Optional { data_type }) => data_type.as_ref().all_values(),
+            _ => false,
+        }
+    }
+
+    pub fn possible_values(&self) -> Option<Vec<Value>> {
```

Comment:
Rename all `possible_values` into `values` because those values are the possible values of elements in the set, not of the set itself.

Furthermore it is more consistent with the `into_values` method